spec/game_spec.rb

```diff
@@ -0,0 +1,136 @@
+require 'game'
+
+describe Game do
+  it 'can be initialized' do
+    expect(Game).to respond_to(:new)
+  end
+
+  it 'initializes with starting conditions' do
+    expect(subject.current_turn).to eq 0
+    expect(subject.game_score).to eq 0
+    expect(subject.current_frame).to eq nil
+    expect(subject.frames).to eq []
+  end
+
+  describe '#start_new_frame' do
+    it 'begings the game and proceeds to first frame' do
+      subject.start_new_frame
+      expect(subject.current_turn).to eq 1
+      expect(subject.current_frame).to be_kind_of Frame
+    end
+  end
+
+  describe '#roll_current_frame' do
+    before(:each) do
+      subject.start_new_frame
+    end
+
+    it 'updates the current frame based on rolls' do
+      subject.roll_current_frame(5, 3)
+      expect(subject.current_frame.throws).to eq [5, 3]
+    end
```

Comment:
This test is not isolated. In fact, you rely heavily on both classes being present for this to work. It's ok for a feature test, but not a unit test.
So maybe you want to duplicate this file, name one "feature test" that will test the whole program working together, and then isolate the one that stays a unit test from Frame.
Where isolating becomes difficult is when two classes are heavily coupled, which is currently the case.

When that happens, you want to:
- define the overall user interface
-use it to write good feature tests, from the point of view of the user (so not feature tests that look into the state, but rather, user rolls a full game, and then checks the score).
- these tests should not change through the whole refactor, given that the functionality to the user should stay the same
- then start moving things in different classes, relying on unit test to make each change.